### PR TITLE
server: Fix message when corrupted or unsupported format is found.

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
This is a fix to provide a little more elaborate message.

## Motivation and Context
Refer https://github.com/minio/minio/issues/4140

## How Has This Been Tested?
Manually.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/storage-rpc-client.go [hunk 1]
```diff
@@ -116,7 +116,8 @@ func newStorageRPC(endpoint Endpoint) StorageAPI {
 // Stringer interface compatible representation of network device.
 func (n *networkStorage) String() string {
 	// Remove the storage RPC path prefix, internal paths are meaningless.
-	serviceEndpoint := strings.TrimPrefix(n.rpcClient.ServiceEndpoint(), storageRPCPath)
+	serviceEndpoint := strings.TrimPrefix(n.rpcClient.ServiceEndpoint(),
+		path.Join(minioReservedBucketPath, storageRPCPath))
 	return n.rpcClient.ServerAddr() + ":" + serviceEndpoint
```
Comment (krisis) [inline cmd/storage-rpc-client.go]: Could you add a unit test to test `String()` function of network storage for a bunch of endpoint paths?

File: cmd/prepare-storage.go [hunk 2]
```diff
@@ -229,7 +229,11 @@ func retryFormattingXLDisks(firstDisk bool, endpoints EndpointList, storageDisks
 			// most part unless one of the formats is not consistent
 			// with expected XL format. For example if a user is trying
 			// to pool FS backend.
-			if err := checkFormatXLValues(formatConfigs); err != nil {
+			if index, err := checkFormatXLValues(formatConfigs); err != nil {
+				if retryCount < 5 {
```
Comment (krisis) [inline cmd/prepare-storage.go]: We should use a `const` instead of a `int` literal.

File: cmd/storage-rpc-client.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/storage-rpc-client.go]: Sure.

File: cmd/prepare-storage.go (hunk 2 above)
Comment (harshavardhana) [reply cmd/prepare-storage.go]: Sure!

File: cmd/prepare-storage.go [hunk 3]
```diff
@@ -219,17 +222,22 @@ func retryFormattingXLDisks(firstDisk bool, endpoints EndpointList, storageDisks
...
-			// with expected XL format. For example if a user is trying
-			// to pool FS backend.
-			if err := checkFormatXLValues(formatConfigs); err != nil {
+			// with expected XL format. For example if a user is
+			// trying to pool FS backend.
+			if index, err := checkFormatXLValues(formatConfigs); err != nil {
+				if retryCount > maxRetryAttempts {
+					errorIf(err, "Detected disk (%s) in unexpected format",
+						storageDisks[index])
+					continue
```
Comment (vadmeste) [inline cmd/prepare-storage.go]: Why we 'continue' here ? Shouldn't we should immediately quit if format xl format is corrupted.

File: cmd/storage-rpc-client.go [hunk 4]
```diff
@@ -116,8 +116,13 @@ func newStorageRPC(endpoint Endpoint) StorageAPI {
 // Stringer interface compatible representation of network device.
 func (n *networkStorage) String() string {
 	// Remove the storage RPC path prefix, internal paths are meaningless.
-	serviceEndpoint := strings.TrimPrefix(n.rpcClient.ServiceEndpoint(), storageRPCPath)
-	return n.rpcClient.ServerAddr() + ":" + serviceEndpoint
+	serviceEndpoint := strings.TrimPrefix(n.rpcClient.ServiceEndpoint(),
```
Comment (vadmeste) [inline cmd/storage-rpc-client.go:119]: This looks like needed to be sent in a separate PR. Also no change in rpc server to support the new rpc path ?

Comment (harshavardhana) [reply cmd/storage-rpc-client.go:119]: The path is only canonicalized here. Server doesn't need any change. See this is a String method.

File: cmd/prepare-storage.go (hunk 3 above)
Comment (harshavardhana) [reply cmd/prepare-storage.go]: We do is to avoid timing issue of how we need to deal with servers coming online. See the original issue.

Comment (vadmeste) [reply cmd/prepare-storage.go]: yes, I understand, could you add a comment?

Comment (harshavardhana) [reply cmd/prepare-storage.go]: Sure..

**Review by vadmeste — APPROVED** [review]

**Review by krisis — APPROVED** [review]
